app/app.go

```diff
@@ -62,3 +69,13 @@ func New(ds store.DataStore, config ...Config) App {
 func (a *app) HealthCheck(ctx context.Context) error {
 	return a.store.Ping(ctx)
 }
+
+func (a *app) ProvisionDevice(ctx context.Context, dev model.NewDevice) error {
+	return a.store.InsertDevice(ctx, model.Device{
+		ID:        dev.ID,
+		UpdatedTS: time.Now(),
+	})
```

Comment:
As we already know distributed systems are complicated, and we can occasionally lose workflow executions or potentially retry execution of tasks, I'd suggest to apply the upsert logic when provisioning a device instead of inserting.

Reply:
Hmm... I think it's more correct to do an insert for the provision endpoint, but I see that I should detect duplicate key errors and return 409 (conflict) response in these cases. This way we can make this response an exception in the workflow.